src/api/v4/dataview/histogram.js

```diff
@@ -0,0 +1,186 @@
+var _ = require('underscore');
+var Base = require('./base');
+var HistogramDataviewModel = require('../../..//dataviews/histogram-dataview-model');
+
+/**
+ * Histogram dataview object
+ *
+ * @param {carto.source.Base} source - The source where the dataview will fetch the data.
+ * @param {string} column - The column name to get the data.
+ * @param {object} options
+ * @param {number} [options.bins=10] - Number of bins to aggregate the data range into.
+ *
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function Histogram (source, column, options) {
+  this._initialize(source, column, options);
+  this._bins = this._options.bins;
+}
+
+Histogram.prototype = Object.create(Base.prototype);
+
+Histogram.prototype.DEFAULTS = {
+  bins: 10
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {HistogramData}
+ * @api
+ */
+Histogram.prototype.getData = function () {
+  if (this._internalModel) {
+    return this._parseData(this._internalModel.get('data'), this._internalModel.get('nulls'), this._internalModel.get('totalAmount'));
+  }
+  return null;
+};
+
+/**
+ * Return the totals data (not affected by filters)
+ *
+ * @return {HistogramData}
+ * @api
+ */
+
+Histogram.prototype.getTotalsData = function () {
```

Comment:
Totals are the whole data not affected by filters. The grey bars in our current widgets. They serve as comparison of the current histogram view against the total one.

In our first version of public API the only difference will be one being affected by view bounding box and the other one showing totals.

Reply:
what about `getUnfiltered` data ?

Reply:
We renamed it while doing the time-series milestone because we understood it better this way. Perhaps it was because in our widgets the option to show it is `show totals` not `show unfiltered data`.

Reply:
I would personally remove this for now because:

1. The fact that BUILDER needed this doesn't mean every developer out there will need it. In case they do, let's find the best way to do it (we don't have to propagate the not so good decissions that we made internally like having dataviews with many responsibilities).

2. Wouldn't someone be able to accomplish this exact thing by using another unfiltered dataview?